packages/edit-site/src/components/page-templates/index.js

```diff
@@ -92,11 +93,22 @@ export default function PageTemplates() {
 		};
 	}, [ layout, activeView ] );
 	const [ view, setView ] = useState( defaultView );
+
+	// Sync the layout from the URL to the view state.
+	const onChangeUrlLayout = useEvent( () => {
+		setView( ( currentView ) => ( {
+			...currentView,
+			type: layout ?? DEFAULT_VIEW.type,
+		} ) );
+	} );
```

Comment:
Why not use the `set` state class inside the effects? I don't see the need for `useEvent` here.

Reply:
The effect here is indeed not needed. I've used it more as a way to prevent future issues, folks randomly adding dependencies to these effects and clarify the flow.

Reply:
> I've used it more as a way to prevent future issues, folks randomly adding dependencies to these effects and clarifying the flow.

It seems like an odd pattern to prevent possible future issues, which are hard to avoid anyway 😅 . It might be better if we express this via an inline comment note.

I trust your judgment here.

Reply:
I can remove for now, it's ok. I'm not sure anyway :P